client/task-list/tasks/payments/PaymentGatewaySuggestions/components/Setup/Connect.js

```diff
@@ -94,22 +95,24 @@ export const Connect = ( {
 	const helpText = setupHelpText && (
 		<p dangerouslySetInnerHTML={ sanitizeHTML( setupHelpText ) } />
 	);
-	const DefaultForm = ( props ) => (
-		<DynamicForm
-			fields={ fields }
-			isBusy={ isUpdating }
-			onSubmit={ handleSubmit }
-			submitLabel={ __( 'Proceed', 'woocommerce-admin' ) }
-			validate={ ( values ) => validateFields( values, fields ) }
-			{ ...props }
-		/>
-	);
+	const DefaultForm = useRef( ( props ) => {
+		return (
+			<DynamicForm
+				fields={ fields }
+				isBusy={ isUpdating }
+				onSubmit={ handleSubmit }
+				submitLabel={ __( 'Proceed', 'woocommerce-admin' ) }
+				validate={ ( values ) => validateFields( values, fields ) }
+				{ ...props }
+			/>
+		);
+	} );
```

Comment:
This way we don't re-create the DynamicForm component every time the render method is being called.